### PR TITLE
Ponytail audit (−10.4k lines) + worker-restart hardening + deterministic dependency closure

### PR DESCRIPTION
**Pull request by thedotmack**
## What's in here

This branch bundles several lines of work that all land together because they're built, synced, and validated as one runtime. Headline is a repo-wide ponytail (anti-over-engineering) audit, on top of worker-restart hardening, a deterministic dependency closure, and assorted fixes.

### Ponytail audit — ~−10,443 lines (src + tests), 0 behavior change

Executed as 8 self-contained slices, each its own reviewable commit, each gated: pre-delete grep → `tsc --noEmit` clean → `bun test` green → commit. Final state: **`tsc` clean, full suite 2080 pass / 0 fail, dead-reference sweep zero, worker boots clean.**

| | files | + | − | net |
|---|---|---|---|---|
| `src/` | 136 | 953 | 7,558 | **−6,605** |
| `tests/` | 45 | 944 | 4,782 | **−3,838** |
| total | 181 | 1,897 | 12,340 | **−10,443** |

- **Slice 0** — remove commit-hash fabrication verification (greedy-regex false positives cost more than the edge case it caught)
- **Slice 1** — delete the duplicate SQLite stack (`ClaudeMemDatabase`/`MigrationRunner` + a parallel free-function CRUD API only tests used); rewrite tests against the real `SessionStore` path (~−3,260 src)
- **Slice 6** — dead storage methods; `EnvManager` 3× credential blocks → one `CREDENTIAL_KEYS` loop; `util.parseEnv` + `AbortSignal.timeout` swaps
- **Slice 4** — `src/server` sweep: dead `outbox.ts`, `mcp/` surface, always-inert provider/event boundaries; hoisted duplicate auth helpers
- **Slice 7** — `toError()` helper; dead ProcessManager/ModeManager/AgentFormatter fns; 3 dead barrels
- **Slice 5** — orphan `src/bin` + `src/adapters`; dead cursor-utils/logger/UI-hook code
- **Slice 2** — integrations/sync dead exports + Chroma-write-path cleanup (the plan's 5-installer consolidation was descoped after a feasibility analysis: 5 divergent config schemas + thin test coverage made a 5→1 collapse a line-count trophy with real regression risk)
- **Slice 3** — worker dead-code + `OpenAICompatibleProvider` base (Gemini/OpenRouter twins) + SearchManager dedup. Both behavior-preserving refactors **adversarially reviewed → behavior preserved**.

Verify-before-cut caught several false-dead items that would have broken the build and were kept: the `ServerV1Routes runtime?` option (live smoke test), the `Bun.which` swap (npx-cli runs under Node), `Middleware.ts`/`context-generator` (live consumers), `SSEEventPayload` + `RateLimitStore.get` (live, hidden by the plan's grep exclusions).

### Worker restart — single source of truth (7-phase)

Self-replacing worker (old worker spawns its successor), restart proves itself or exits 1, one spawn gate, PID file demoted (health + port are the liveness oracle), deleted the sync-script cache-mirror and HTTP restart trigger.

### Deterministic plugin runtime dependency closure (plan-10)

Ship a deterministic runtime dependency closure; fail loud at install time on a broken closure; clean-room install + import smoke test as a backstop; CI/publish gated on it; `gen-plugin-lockfile` degrades gracefully when bun is absent.

### Fixes & maintenance

- `fix(sdk)`: a stale Claude CLI can no longer silently kill every observation
- `fix(settings)`: bootstrap notices go to stderr, never stdout
- `test`: no test ever touches the real `~/.claude-mem` again
- `chore(deps)`: daily upgrade pass (agent SDK, better-auth, posthog-node, dompurify)
- plugin artifacts rebuilt from post-audit source

## Validation
- `bunx tsc --noEmit` → exit 0
- `bun test` → 2080 pass / 18 skip / 0 fail
- `npm run build-and-sync` → worker restart verified, healthy on port 37777

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by thedotmack** [comment]
@greptile review

File: src/shared/EnvManager.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 
 import { existsSync, readFileSync, writeFileSync, mkdirSync, chmodSync } from 'fs';
+import { parseEnv } from 'util';
 import { logger } from '../utils/logger.js';
```
[the remark this answers is not in the document]
Comment (thedotmack) [reply src/shared/EnvManager.ts:4]: Fixed in 5b2d1e3b2. Bumped `engines.node` to `>=20.12.0` (the real floor for `util.parseEnv`) in both the root npm package and the generated `plugin/package.json` — the latter via its generator in `scripts/build-hooks.js` so rebuilds stay consistent. Verified this is the sole version-gating API added in the PR (full 189-file diff sweep: no other API requires >20.12). Matches the audit's intent (deliberately adopted stdlib `util.parseEnv`; the code comment already declared "Node >=20.12").
